src/edgepi/calibration/protobuf_mapping.py

```diff
@@ -0,0 +1,78 @@
+"""module to map protobuf data to a class"""
+
+from dataclasses import dataclass
+from edgepi.calibration.eeprom_mapping_pb2 import EepromLayout
+from edgepi.calibration.calibration_constants import CalibParam
+
+@dataclass
+class Keys:
+    """
+    Dataclass to store key strings
+    """
+    certificate: str = None
+    private: str = None
+
+
+class EdgePiEEPROMData:
+    # pylint: disable=too-many-instance-attributes
+    """
+    Dataclass to store edgepi reserved values
+    dac_calib_parms (dict): list of calibration parameters
+    adc_calib_parms (dict): list of calibration parameters
+    rtd_calib_parms (dict): list of calibration parameters
+    tc_calib_parms (dict): list of calibration parameters
+    config_key (Keys): dataclass
+    data_key (Keys): dataclass
+    serial (str)
+    model (str)
+    client_id (str)
+    """
+    def __init__(self, data_to_unpack: EepromLayout = None):
+        self.dac_calib_params = self.calib_message_to_dict(data_to_unpack.dac)
+        self.adc_calib_params = self.calib_message_to_dict(data_to_unpack.adc)
+        self.rtd_calib_params = self.calib_message_to_dict(data_to_unpack.rtd)
+        self.rtd_hw_params = self.hw_message_to_dict(data_to_unpack.rtd)
+        self.tc_calib_params = self.calib_message_to_dict(data_to_unpack.tc)
+        self.tc_hw_params = self.hw_message_to_dict(data_to_unpack.tc)
+        self.config_key = self.keys_to_dataclass(data_to_unpack.config_key)
+        self.data_key = self.keys_to_dataclass(data_to_unpack.data_key)
+        self.serial = data_to_unpack.serial_number
+        self.model = data_to_unpack.model
+        self.client_id = data_to_unpack.client_id
+
+    def calib_message_to_dict(self, data_to_unpack: EepromLayout = None):
+        """
+        Function to unpack message to list
+        Args:
+            data_to_unpack: EepromLayout message modules
+        Returns:
+            calib_dict: calib param to dictionary
+        """
+        calib_dict={}
+        for indx, ch in enumerate(data_to_unpack.calibs):
+            calib_dict[indx] = CalibParam(gain=ch.gain,
+                                        offset=ch.offset)
+        return calib_dict
```

Comment:
What data type is `data_to_unpack` here? It seems you're passing an iterable instead of `EepromLayout`.

Reply:
I'm passing one of the message type inside the protobuf module